docs/design/solutions/pce-exclude-subgraph/pce-exclude-subgraph.md

```diff
@@ -0,0 +1,19 @@
+# Ability to exclude given graph from path computation
+
+## Goals
+Add general excluding mechanics in Computation Engine. It will be used in scenarios like:
+- Calculate a diverse path
+- Evacuate flows from ISL/Switch (reroute with exclude)
+- Allow user to specify exclusion list from API
+
+## Solution overview
+There are two rules of excluding
+- Exclude specified ISL
+- Exclude entire switch
```

Comment:
Excluding switch you exclude all ISL attached to this switch. So switch exclusion is equal to ISL exclusion.

In other word we need to implement exclusion of N ISL from path computation.

Reply:
I'd vote for keeping switch filter and do a performance test afterward, so we can decide later. Having this filtering logic inside PCE will simplify external business logic a bit.

Reply:
We can and should have "exclude switch" into NB API.

The possibility to convert "exclude switch" into "exclude N ISL" is our implementation details and external world should not worry/know about them.